### ISSUE TITLE
Impossible to delete a todo in the "MongoDB To-Do List" tutorial

### ISSUE THREAD

**Issue by LoicPoullain** [FoalTS/foal#280]
**Describe the bug**

When following [this tutorial](https://foalts.gitbook.io/docs/tutorials/mongodb-to-do-list/2-with-typeorm) the server responds with a 400 error upon todo deletion.

**To Reproduce**
1. Follow the "MongoDB To-Do List" tutorial.
2. Try to delete a random todo by clicking on a checkbox.

**Expected behavior**

The todo should be deleted.

**Screenshots**

<img width="588" alt="capture d ecran 2018-12-10 a 21 49 33" src="https://user-images.githubusercontent.com/13604533/49761561-c6df1400-fcc7-11e8-9b04-343048d512d0.png">

**Comment by LoicPoullain** [comment]
Origin of the error : MongoDB ids are not numbers. When requesting `DELETE /todos/:id`, `Ajv` tries to convert the id to a number (`id: { type: 'number' }`) which makes the request to fail.

A first fix is to replace `number` by `string` in `api.controller.ts`

**Comment by LoicPoullain** [comment]
This first fix leads to another error : 404 - NOT FOUND.

After a first investigation, updating the following lines in `api.controller.ts` and `todo.entity.ts` should fix the issue:

```typescript
import { ObjectId } from 'mongodb';
// ...
const todo = await getMongoRepository(Todo).findOne({ _id: new ObjectId(ctx.request.params.id) });
```

```typescript
import { ObjectID }  from 'typeorm';
// ...
  @ObjectIdColumn()
  id: ObjectID;
```

The main issue so far is that the compilation "fails" since the `_id` attribute does not exist on `Todo` (but `id` does).

**Comment by LoicPoullain** [comment]
Resolved with the new tutorial (see #360)
